src/main/java/com/zendesk/maxwell/replication/MaxwellReplicator.java

```diff
@@ -268,6 +268,12 @@ private RowMapBuffer getTransactionRows() throws Exception {
 						// to us starting on a WRITE_ROWS event -- we sync the schema position somewhere
 						// kinda unsafe.
 						processQueryEvent(qe);
+					} else if (sql.toUpperCase().startsWith("INSERT INTO MYSQL.RDS_HEARTBEAT")) {
+						// RDS heartbeat events take the following form:
+						// INSERT INTO mysql.rds_heartbeat2(id, value) values (1,1483041015005) ON DUPLICATE KEY UPDATE value = 1483041015005
+						// As a result they are processed as query events.
+						// When these occur we need to update to update our position.
+						processRDSHeartbeatEvent(qe);
```

Comment:
I'd just remove all the indirection here and process this inside `processQueryEvent`, unless there's a reason not to.